src/lib.rs

```diff
@@ -1,32 +1,305 @@
+  /// If you used JSON to embed multiple properties (such as FirstName, LastName, Email, etc),
+  /// they can be retrieved here as App.Agent.FirstName, etc. (FIXME)
+  pub static ref APP_AGENT_ID_STR: String = APP_GLOBALS.app_agent_id_str.clone();
+
+  /// The hash of your public key.
+  /// This is your node address on the DHT.
+  /// It can be used for node-to-node messaging with `send` and `receive` functions.
+  pub static ref APP_AGENT_KEY_HASH: HashString = APP_GLOBALS.app_agent_key_hash.clone();
+
+  /// The hash of the first identity entry on your chain (The second entry on your chain).
+  /// This is your peer's identity on the DHT.
+  pub static ref APP_AGENT_INITIAL_HASH: HashString = APP_GLOBALS.app_agent_initial_hash.clone();
+
+  /// The hash of the most recent identity entry that has been committed to your chain.
+  /// Starts with the same value as APP_AGENT_INITIAL_HASH.
+  /// After a call to `update_agent` it will have the value of the hash of the newly committed identity entry.
+  pub static ref APP_AGENT_LATEST_HASH: HashString = APP_GLOBALS.app_agent_latest_hash.clone();
+}
+
+
+//--------------------------------------------------------------------------------------------------
+// SYSTEM CONSTS
+//--------------------------------------------------------------------------------------------------
+
+// HC.Version
+const VERSION : u16 = 1;
+const VERSION_STR : &'static str = "1";
+
+
+// HC.HashNotFound
+// FIXME keep in sync with HcApiReturnCode?
+pub enum ErrorCode {
+  FunctionNotImplemented,
+  HashNotFound,
+}
+
 
-type Hash = String;
+// HC.Status
+// WARNING keep in sync with CRUDStatus
+bitflags! {
+  pub struct EntryStatus: u8 {
+    const LIVE     = 1 << 0;
+    const REJECTED = 1 << 1;
+    const DELETED  = 1 << 2;
+    const MODIFIED = 1 << 3;
+  }
+}
 
-pub mod anchors;
 
-pub fn commit<S: Into<String>>(entry_type: S, entry: serde_json::Value) -> String {
-    "".to_string()
+// HC.GetMask
+bitflags! {
+  pub struct GetEntryMask: u8 {
+    const ENTRY      = 1 << 0;
+    const ENTRY_TYPE = 1 << 1;
+    const SOURCES    = 1 << 2;
+  }
+}
+// explicit `Default` implementation
+impl Default for GetEntryMask {
+    fn default() -> GetEntryMask {
+        GetEntryMask::ENTRY
+    }
 }
 
-pub fn link<S: Into<String>>(base: Hash, tag: S, target: Hash) {
 
+// HC.LinkAction
+pub enum LinkAction {
+    ADD,
+    DELETE,
 }
 
-pub fn get(entry_hash: Hash) -> Option<String> {
-    None
+
+// HC.PkgReq
+pub enum PkgRequest {
+    CHAIN,
+    CHAIN_OPTION,
+    ENTRY_TYPES,
 }
 
-pub fn get_links<S: Into<String>>(bash: Hash, tag: S) -> Vec<Hash> {
-    vec![]
+
+// HC.PkgReq.ChainOpt
+pub enum ChainOption {
+    NONE,
+    HEADERS,
+    ENTRIES,
+    FULL,
 }
 
 
+// HC.Bridge
+pub enum BridgeSide {
+    FROM,
+    TO,
+}
 
 
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        assert_eq!(2 + 2, 4);
+// HC.SysEntryType
+// WARNING Keep in sync with SystemEntryType in holochain-rust
```

Comment:
Now that I'm seeing the pattern, I'm wondering actually, how do we prevent from duplication of these constants in the first place. Is there no good way to boil down to a single source of truth?

Reply:
Maybe be could with bindings, especially for the Rust version of Zome API. The point is to have clear Zome API separate from Holochain library though.
The comments are for us holochain devs, not end-users.